src/map.rs

```diff
@@ -382,6 +436,47 @@ impl<K, V, S> HashMap<K, V, S> {
             }
         }
     }
+
+    /// Presize the table to accommodate the given number of elements.
+    fn presize(&mut self, size: usize) {
+        // NOTE: this is a stripped-down version of try_presize for use only when we _know_ that
+        // the table is new, and that therefore we won't have to help out with transfers or deal
+        // with contending initializations.
+
+        // safety: we are creating this map, so no other thread can access it,
+        // while we are initializing it.
+        let guard = unsafe { epoch::unprotected() };
+
+        let requested_capacity = if size >= MAXIMUM_CAPACITY / 2 {
+            MAXIMUM_CAPACITY
+        } else {
+            // round the requested_capacity to the next power of to from 1.5 * size + 1
+            // TODO: find out if this is neccessary
+            let size = size + (size >> 1) + 1;
+
+            std::cmp::min(MAXIMUM_CAPACITY, size.next_power_of_two())
+        } as usize;
+
+        // sanity check that the map has indeed not been set up already
+        assert_eq!(self.size_ctl.load(Ordering::SeqCst), 0);
+        assert!(self.table.load(Ordering::SeqCst, &guard).is_null());
```

Comment:
Should we add a short note here about what would need to be done if at some point we allow a non-zero minimum capacity? Just something that makes clear that if `with_capacity` fails the `size_ctl` assertion it's not because it _has_ to always be 0, but because it is not handled.

Reply:
I'm not sure I follow? This method should only ever be called if the map has not yet been allocated, at which point `size_ctl` should indeed always be zero. Or am I missing something?

Reply:
`size_ctl` is _currently_ always zero, but it doesn't have to be. In general, it stores the minimum capacity allocated for the table. This value is checked in `try_presize` and overrides `DEFAULT_CAPACITY` in `init_table` if `> 0`.

I'm not saying we have to implement this behaviour for `presize` now, I would assume `size_ctl` to stay `0`. But if someone were to change this in the future, this assertion makes it look a lot like it _must_ be `0` if we use `presize`. It's not _that_ important, I already approved the changes themselves 😉